src/docker.ts

```diff
@@ -57,8 +57,28 @@ export async function push(imageName: string, auth: string): Promise<void> {
 }
 
 export function tag(existingImageName: string, newImageName: string): void {
-  const tagOutput = spawnSync('docker', ['tag', existingImageName, newImageName], { encoding: 'utf-8', stdio: 'inherit' })
-  if (tagOutput.status !== 0) {
-    throw new Error(tagOutput.error?.message ?? 'Error tagging image')
+  spawnDockerSync(['tag', existingImageName, newImageName], { stdio: 'inherit' })
+}
+
+function spawnDockerSync(args: string[], options?: { stdio?: StdioOptions }): SpawnSyncReturns<string> {
+  const opts: SpawnSyncOptionsWithStringEncoding = { encoding: 'utf-8', ...options }
+  const result = spawnSync('docker', args, opts)
+  const exitCode = result.status
+  if (
+    (exitCode !== null && exitCode !== 0) ||
+    result.error !== undefined
+  ) {
+    if (result.stderr?.includes('No such image')) {
+      throw new Error('Docker failed to find image. Make sure you have built the image and it\'s listed in the `docker images` command output before running this command.')
+    }
+    if (result.stderr?.includes('Cannot connect to the Docker daemon')) {
+      throw new Error('Docker failed to connect to the Docker daemon. Make sure Docker is running and you have permission to access it.')
+    }
+    if (result.error?.message.includes('ENOENT')) {
+      throw new Error('Docker command not found. Make sure Docker is installed and in your PATH.')
```

Comment:
can we provide a command to check this?

Reply:
If we hit this, then we know the `docker` command isn't in the PATH. Do you think we should point to Docker's docs here? Like maybe [this page](https://docs.docker.com/get-docker/). We could also say that they should be able to run `docker version` successfully.